framework/src/csifw/CSILogsDumper.c

```diff
@@ -0,0 +1,209 @@
+            }
+        }
+
+        // Check if there's data on the FIFO
+        if (FD_ISSET(gFifoFd, &readfds)) {
+            CSIFW_LOGI("Received an activity on Fifo descriptors");
+
+            if (!gWaitingForData) {
+                // Get the length from the first 2 bytes
+                len = recv_message(gFifoFd, (void *)gCSIDataDumpBuffer, 2);
+                if (len <= 0) {
+                    CSIFW_LOGE("operation to read length of data from FIFO Failed");
+                    continue;
+                }
+                uint16_t length;
+                memcpy(&length , gCSIDataDumpBuffer, sizeof(length));
+                CSIFW_LOGI("Read %d length to client:", length);
+                gRemainingBytes = length;
+                gWaitingForData = 1;
+                gBytesReadSofar = 0;
+            }
+            if (gWaitingForData) {
+                // We are expecting to read the actual data
+                ssize_t bytes_read = read(gFifoFd, gCSIDataDumpBuffer + gBytesReadSofar, gRemainingBytes);
+                if (bytes_read < 0) {
+                    int err_no = errno;
+                    if (err_no == EAGAIN || err_no == EINTR) {
+                        continue;
+                    }
+                    CSIFW_LOGE("read error %d\n", err_no);
+                    continue; // or return
+                }
+
+                gBytesReadSofar += bytes_read;
+                gRemainingBytes -= bytes_read;
+
+                if (gRemainingBytes == 0) {
+                    // Complete data received
+                    gWaitingForData = 0;
+                    // Send the data to the client
+                    len = sendto(gSockFd, gCSIDataDumpBuffer, gBytesReadSofar, 0, (struct sockaddr *)&gClientAddr, sizeof(gClientAddr));
+                    if (len == -1) {
+                        CSIFW_LOGE("sendto error:%d", errno);
+                        continue;
+                    }
+                    CSIFW_LOGI("Sent %d bytes to client", len);
+                    print_buf(gCSIDataDumpBuffer,len);
+                }
+            }
+        }
+    }
+}
+
+CSIFW_RES csi_logs_dumper_init()
+{
+	if ((gSockFd = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
+		CSIFW_LOGE("failed to create a socket fd");
+		return CSIFW_ERROR;
+	}
+
+    memset(&gServerAddr, 0, sizeof(gServerAddr));
+    gServerAddr.sin_family = AF_INET;
+    gServerAddr.sin_addr.s_addr = INADDR_ANY;
+    gServerAddr.sin_port = htons(PORT);
+
+	if (bind(gSockFd, (struct sockaddr*)&gServerAddr, sizeof(gServerAddr)) < 0) {
+		CSIFW_LOGE("binding socket failed");
+        close(gSockFd);
+		return CSIFW_ERROR;
+	}
+    printf("server Socket Created\n");
+    gFifoFd = open(CSI_DUMP_DATA_QUEUE_NAME, O_RDONLY | O_NONBLOCK);
+    if (gFifoFd < 0) {
+        CSIFW_LOGE("open CSI_DUMP_DATA_QUEUE fail %d", errno);
+        close(gSockFd);
+        return CSIFW_ERROR;
+    }
+
+    if (pthread_create(&gThreadId, NULL, dataTransmitter, NULL) != 0) {

```

Comment:
When you create a pthread, you should call pthread_join or pthread_detach. If you don't, it will have a memory leak.